src/main/java/List.java

```diff
@@ -0,0 +1,315 @@
+        final int indexOfEndDatePrefix = line.indexOf("/to");
+        if (indexOfEndDatePrefix > indexOfStartDatePrefix) {
+            eventEndDate = line.substring(indexOfEndDatePrefix).replaceFirst("/to", "").trim();
+        } else {
+            eventEndDate = line.substring(indexOfEndDatePrefix, indexOfStartDatePrefix).replaceFirst("/to", "").trim();
+        }
+
+        dateFieldNotEmpty(eventEndDate);
+
+        return eventEndDate;
+    }
+
+    private String extractEventStartDate(String line) {
+        String eventStartDate;
+        final int indexOfStartDatePrefix = line.indexOf("/from");
+        final int indexOfEndDatePrefix = line.indexOf("/to");
+        if (indexOfStartDatePrefix > indexOfEndDatePrefix) {
+            eventStartDate = line.substring(indexOfStartDatePrefix).replaceFirst("/from", "").trim();
+        } else {
+            eventStartDate = line.substring(indexOfStartDatePrefix, indexOfEndDatePrefix).replaceFirst("/from", "").trim();
+        }
+
+        dateFieldNotEmpty(eventStartDate);
+
+        return eventStartDate;
+    }
+
+    public void printList() {
+        System.out.println("\tHere are the tasks in your list:");
+        int i = 0;
+        for (Task a: itemArrayList) {
+            System.out.println("\t" + (i + 1) + "." + a);
+            i += 1;
+        }
+    }
+
+    public void markItem(String line) {
+        try {
+            int itemNum = Integer.parseInt(line.substring(5));
+
+            if (itemNum > this.getNumItems() || itemNum <= 0) {
+                printInputIndexOutOfRangeMessage();
+            } else {
+                this.markListItemAsDone(itemNum);
+                printTaskMarkedMessage(itemNum);
+            }
+        } catch (NumberFormatException  e) {
+            printInputIndexNotAnIntegerMessage();
+        } catch (Exception e) {
+            printUnknownErrorMessage();
+        }
+    }
+
+    private static void printUnknownErrorMessage() {
+        System.out.println("Unknown error experienced.");
+    }
+
+    private void printTaskMarkedMessage(int itemNum) {
+        System.out.println("\tNice! I've marked this task as done:");
+        System.out.println("\t  " + itemArrayList.get(itemNum - 1));
+    }
+
+    private static void printInputIndexOutOfRangeMessage() {
+        System.out.println("\tInput index number out of range.");
+    }
+
+    public void unmarkItem(String line) {
+        try {
+            int itemNum = Integer.parseInt(line.substring(7));
+
+            if (itemNum > this.getNumItems() || itemNum <= 0) {
+                printInputIndexOutOfRangeMessage();
+            } else {
+                this.markListItemAsUnDone(itemNum);
+                printTaskUnmarkedMessage(itemNum);
+            }
+        } catch (NumberFormatException  e) {
+            printInputIndexNotAnIntegerMessage();
+        } catch (Exception e) {
+            printUnknownErrorMessage();
+        }
+    }
```

Comment:
Consider catching specific errors inorder to prompt the user with more informative text error. Consider `ArrayIndexOutOfBoundsException`, etc.